### PR TITLE
Fix(i18n): Inaccessible public folder on non-default domains

### LINKED ISSUE #54765: Public files not accessible on i18n domains
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:05 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6000
    Binaries:
      Node: 20.2.0
      npm: 9.6.4
      Yarn: 1.22.19
      pnpm: 8.7.0
    Relevant Packages:
      next: 13.4.20-canary.12
      eslint-config-next: 13.4.19
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.2.2
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Internationalization (i18n), Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue or a replay of the bug

https://app.replay.io/recording/reproduction--e17b7c87-6019-40b2-b719-1e9b9ce80b13

### To Reproduce

- Create new next app
- Add robots.txt in public folder
- Setup basic i18n (see example below)
- Make sure you are on different domains (as example)
- Go to default domain `/robots.txt`
- See the robots.txt output
- Go to different domain `/robots.txt`
- See 404

```js
  i18n: {
    defaultLocale: "en",
    locales: ["en", "nl-NL"],
    domains: [
      {
        domain: "com.rob.ticketswap.pizza",
        defaultLocale: "en",
        http: false,
      },
      {
        domain: "nl.rob.ticketswap.pizza",
        defaultLocale: "nl-NL",
        http: false,
      },
    ],
    localeDetection: false,
  },
```

### Describe the Bug

This bug presented itself specifically in Next v13.

Public files are no longer accessible on all domains that are not the default domain (default locale domain).

### Expected Behavior

All files in public folder should be accessible on every domain that is setup in the i18n config.

### Which browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by pekarja5**
Changes in #52492 made `public` folder accessible only on default domain if using i18n. PR fixes this by making public directory contents accessible from all i18n domains as the original solution did. 

Fixes #54765

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: acf43f4d59a2ba93c80fe8b96874b0fe8a7e4814

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: acf43f4d59a2ba93c80fe8b96874b0fe8a7e4814

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by mayank1513 — APPROVED** [review]

**Comment by bobbonius** [comment]
Hi there! My colleague created the issue you're referencing and I was wondering if I could be of any assistance to get this through? We want to go to next 13 but this issue is holding us back at the moment.

**Comment by pekarja5** [comment]
@bobbonius There is nothing to do apart from waiting for it to get merged. The fix is really simple as it is just one line edit.  It is already approved so i hope it will get merged and released soon.

**Comment by bobbonius** [comment]
> There is nothing to do apart from waiting for it to get merged. The fix is really simple as it is just one line edit. It is already approved so i hope it will get merged and released soon.

Thanks!

**Comment by aledovskikh** [comment]
Hey! When it'll be released? Looks like this is some kind of critical bug

**Review by chrisneven — APPROVED** [review]

**Comment by chrisneven** [comment]
This is blocking us as well at the moment. It would be really nice if we could get this one true!

**Comment by chrisneven** [comment]
@ijjk, maybe you can take a look at this? Since you made the change initially

Thanks in advance!

**Review by timneutkens — CHANGES_REQUESTED** [review]
This PR is missing tests, please have a look at and follow the PR template to ensure the changes can be reviewed correctly.

**Comment by pekarja5** [comment]
@timneutkens Does the test suite even supports testing access to multiple domains? I could not find any test with such functionality.

The goal is to make the test access files in `public` directory from multiple defined locale domains. If anyone has any tips how to make such test feel free to give me some ideas.

**Comment by aledovskikh** [comment]
Is there any chance to fix and merge it to nearest release? 
We can't upgrade NextJS upper than 13.4.7, because of this bug

**Comment by glenngijsberts** [comment]
@pekarja5  could you please update the description of this PR? 

@timneutkens sorry for tagging again, but any chance you can give some guidance on how to test this (personally I don't know)?

**Comment by pekarja5** [comment]
@glenngijsberts Some changes in particular?

**Comment by glenngijsberts** [comment]
@pekarja5 like Tim said, follow the PR template please. You can start creating a PR (without actually creating it) to see what the template is.

**Comment by persocon** [comment]
lol, just upgraded my nextjs version to 14 and now i'm having this exact problem, when will this PR be updated and merged? 🙏

**Review by ijjk — CHANGES_REQUESTED** [review]
Hi, this would match all locales and not jut the locale domains/default locale. Continuing fix/tests in https://github.com/vercel/next.js/pull/60749

**Pull request closed without merging**
